tests/scene/test_camera_3d.h

```diff
@@ -0,0 +1,362 @@
+		CHECK(test_camera->get_h_offset() == h_offset);
+		test_camera->set_v_offset(v_offset);
+		CHECK(test_camera->get_v_offset() == v_offset);
+		test_camera->set_frustum_offset(frustum_offset);
+		CHECK(test_camera->get_frustum_offset() == frustum_offset);
+		test_camera->set_keep_aspect_mode(Camera3D::KeepAspect::KEEP_HEIGHT);
+		CHECK(test_camera->get_keep_aspect_mode() == Camera3D::KeepAspect::KEEP_HEIGHT);
+		test_camera->set_keep_aspect_mode(Camera3D::KeepAspect::KEEP_WIDTH);
+		CHECK(test_camera->get_keep_aspect_mode() == Camera3D::KeepAspect::KEEP_WIDTH);
+	}
+
+	SUBCASE("Projection mode") {
+		test_camera->set_projection(Camera3D::ProjectionType::PROJECTION_ORTHOGONAL);
+		CHECK(test_camera->get_projection() == Camera3D::ProjectionType::PROJECTION_ORTHOGONAL);
+		test_camera->set_projection(Camera3D::ProjectionType::PROJECTION_PERSPECTIVE);
+		CHECK(test_camera->get_projection() == Camera3D::ProjectionType::PROJECTION_PERSPECTIVE);
+	}
+
+	SUBCASE("Helper setters") {
+		constexpr float fov = 90.0f, size = 6.0f;
+		constexpr float near1 = 0.1f, near2 = 0.5f;
+		constexpr float far1 = 1001.0f, far2 = 1005.0f;
+		test_camera->set_perspective(fov, near1, far1);
+		CHECK(test_camera->get_projection() == Camera3D::ProjectionType::PROJECTION_PERSPECTIVE);
+		CHECK(test_camera->get_near() == near1);
+		CHECK(test_camera->get_far() == far1);
+		CHECK(test_camera->get_fov() == fov);
+		test_camera->set_orthogonal(size, near2, far2);
+		CHECK(test_camera->get_projection() == Camera3D::ProjectionType::PROJECTION_ORTHOGONAL);
+		CHECK(test_camera->get_near() == near2);
+		CHECK(test_camera->get_far() == far2);
+		CHECK(test_camera->get_size() == size);
+	}
+
+	SUBCASE("Doppler tracking") {
+		test_camera->set_doppler_tracking(Camera3D::DopplerTracking::DOPPLER_TRACKING_IDLE_STEP);
+		CHECK(test_camera->get_doppler_tracking() == Camera3D::DopplerTracking::DOPPLER_TRACKING_IDLE_STEP);
+		test_camera->set_doppler_tracking(Camera3D::DopplerTracking::DOPPLER_TRACKING_PHYSICS_STEP);
+		CHECK(test_camera->get_doppler_tracking() == Camera3D::DopplerTracking::DOPPLER_TRACKING_PHYSICS_STEP);
+		test_camera->set_doppler_tracking(Camera3D::DopplerTracking::DOPPLER_TRACKING_DISABLED);
+		CHECK(test_camera->get_doppler_tracking() == Camera3D::DopplerTracking::DOPPLER_TRACKING_DISABLED);
+	}
+
+	memdelete(test_camera);
+}
+
+TEST_CASE("[SceneTree][Camera3D] Position queries") {
+	// Cameras need a viewport to know how to compute their frustums, so we make a fake one here.
+	Camera3D *test_camera = memnew(Camera3D);
+	SubViewport *mock_viewport = memnew(SubViewport);
+	// 4:2.
+	mock_viewport->set_size(Vector2(400, 200));
+	SceneTree::get_singleton()->get_root()->add_child(mock_viewport);
+	mock_viewport->add_child(test_camera);
+	test_camera->set_keep_aspect_mode(Camera3D::KeepAspect::KEEP_WIDTH);
+	REQUIRE_MESSAGE(test_camera->is_current(), "Camera3D should be made current upon entering tree.");
+	SUBCASE("Orthogonal projection") {
```

Comment:
```suggestion

	SUBCASE("Orthogonal projection") {
```
I'd separate the cases from other code as well to make it feel less cluttered

Reply:
Fixed in the newest version. As always, please let me know if I missed anything.